docs/data-model.md

```diff
@@ -0,0 +1,117 @@
+
+Essentially, each alert indicates to us that a new transient has been
+discovered (if nothing was present at that location in the reference image) or
+a known star is varying in brightness.
+
+An alert always includes an object ID, along with a candidate ID,
+which serves as the unique identifier for that alert.
+
+Alerts from different surveys will have different schemas,
+but they have a few common properties:
+
+- Object ID
+- Candidate ID
+- Right ascension (RA)
+- Declination (dec)
+- Brightness and associated error (in magnitude and/or flux-space)
+- New, reference,
+  and difference images (angular and pixel size varying per survey)
+- Various quality flags and metadata about the associated alert
+
+## Catalog
+
+A catalog is a dataset from a survey.
+Internally, a catalog maps one-to-one with a MongoDB collection.
+BOOM stores both _archival_ and _live_ catalogs,
+the former being static sets of objects detected by a survey.
+Archival catalogs will only contain objects,
+but live catalogs will come in sets of three with different data types:
+
+- Alerts (typically named like `{survey_name}_alerts`,
+  with candidate ID as the unique identifier)
+- Objects (typically named like `{survey_name}_alerts_aux`)
+- Images (typically named like `{survey_name}_alerts_cutouts`)
+
+## Alert input stream
+
+A live survey has one or more alert input streams to which BOOM can subscribe.
+Each alert stream contains a different
+subset of alerts and has its own access restrictions.
+For example,
+ZTF has a public stream, a partnership stream, and a Caltech-only stream.
+
+## Cross-match
+
+A cross-match relates an object's ID in one survey to its ID in another
+based on location.
+It's possible that two distinct objects could be detected at the same location,
+due to positional uncertainty, distance, and resolution.
+BOOM cross-matches object IDs in live catalogs with archival catalogs
+the first time an object shows up in an alert, and only that first time.
+
+## Filter
+
+Users declare filters to define what alerts they would like BOOM to pass
+through to an output stream,
+since different users are interested in different types of objects.
+Since surveys can produce millions of alerts per night,
+having filters in place is critical to reduce the effort of detecting
+scientifically relevant objects.
+
+Filters can make use of the cross-matching information since BOOM performs
+cross-matching before applying filters.
+Filters can also make use of light curve data from other live surveys,
+which is as far as we know not possible with any other alert broker.
+
+Each filter is associated with one alert input stream
+and one alert output stream.
+Each alert output stream has its own credentials for authorization.
+These can be used, for example, to connect a Marshal like SkyPortal.
+
+Admin users are able to modify any filter in the system,
+but regular users are only able to modify filters they created.
```

Comment:
I assume this is how things will need to be, but correct me if we need something more complex, e.g., filters owned by user groups or something. From what I've heard from @Theodlz, users aren't doing a whole lot of filter modifications on their own, but they might be in the future with the GUI.

Reply:
Yeah we don't know exactly where the filter UI will be implemented, but my guess is in skyportal which will communicate with boom using some admin token. So let's not worry about this too much, we could skip the permission detail here until we actually implement something related to that